cluster-controller/src/test/java/io/strimzi/controller/cluster/operations/cluster/KafkaClusterOperationsTest.java

```diff
@@ -81,13 +113,76 @@ public void testCreateCluster(TestContext context) {
                 .endMetadata()
                 .withData(cmData)
                 .build();
-        kubeClient.configMaps().create(clusterCm);
+        when(mockCmOps.get(clusterCmNamespace, clusterCmName)).thenReturn(clusterCm);
+        ArgumentCaptor<Service> serviceCaptor = ArgumentCaptor.forClass(Service.class);
+        when(mockServiceOps.create(serviceCaptor.capture())).thenReturn(Future.succeededFuture());
+        ArgumentCaptor<StatefulSet> ssCaptor = ArgumentCaptor.forClass(StatefulSet.class);
+        when(mockSsOps.create(ssCaptor.capture())).thenReturn(Future.succeededFuture());
+
+        KafkaClusterOperations ops = new KafkaClusterOperations(vertx, isOpenShift,
+                mockCmOps,
+                mockServiceOps, mockSsOps,
+                mockPvcOps);
 
         // Now try to create a KafkaCluster based on this CM
         Async async = context.async();
         ops.create(clusterCmNamespace, clusterCmName, createResult -> {
             context.assertTrue(createResult.succeeded());
+            // Assertions about the CM, Services and SS
+
+            // No metrics config  => no CMs created
+            verify(mockCmOps, never()).create(any());
+
+            // We expect a headless and headful service
+            List<Service> capturedServices = serviceCaptor.getAllValues();
+            context.assertEquals(2, capturedServices.size());
+            int headfulIndex = (clusterCmName+"-kafka").equals(capturedServices.get(0).getMetadata().getName()) ? 0 : 1;
+
+            // Assertions on the external service
+            Service headful = capturedServices.get(headfulIndex);
```

Comment:
Are these asserts needed here? Wouldn't it be better to test the details of the Kubernetes resources in test for `KafkaCluster` class?

Reply:
Excellent point. Thanks!